### PR TITLE
Fetch execution outcome when querying transaction status

### PR DESCRIPTION
**Pull request by bowenwang1996**
When we query transaction status from a node currently, it might fail because the node doesn't track all shards and therefore does not know about the execution outcome of some receipts. This PR adds the fetching request from chunk producers for shard that the receipt goes to when such scenario happens.

### CODE REVIEW

File: chain/client/src/view_client.rs [hunk 1]
```diff
@@ -131,8 +137,32 @@ impl ViewClientActor {
...
         if has_tx_result {
             let tx_result = self.chain.get_final_transaction_result(&tx_hash)?;
+            match tx_result.status {
+                FinalExecutionStatus::NotStarted | FinalExecutionStatus::Started => {
+                    for receipt_view in tx_result.receipts.iter() {
+                        let dst_shard_id = *self
+                            .chain
+                            .get_shard_id_for_receipt_id(&receipt_view.id)
+                            .map_err(|e| e.to_string())?;
+                        if self.chain.get_chunk_extra(&head.last_block_hash, dst_shard_id).is_err()
```
Comment (SkidanovAlex) [inline chain/client/src/view_client.rs]: Don't use `is_err` without checking that the error is `DbNotFound`

File: chain/client/src/view_client.rs [hunk 2]
```diff
@@ -131,8 +137,32 @@ impl ViewClientActor {
...
+                        {
+                            let validator = self
+                                .chain
+                                .find_validator_for_forwarding(dst_shard_id)
+                                .map_err(|e| e.to_string())?;
+                            self.receipt_outcome_requests.cache_set(receipt_view.id, ());
+                            self.network_adapter.send(NetworkRequests::ReceiptOutComeRequest(
```
Comment (SkidanovAlex) [inline chain/client/src/view_client.rs]: Add some logic that will prevent spamming with requests

File: chain/client/src/view_client.rs [hunk 3]
```diff
@@ -253,6 +283,49 @@ impl Handler<NetworkViewClientMessages> for ViewClientActor {
...
+                                if shard_id < block.chunks.len() as u64 {
+                                    if verify_path(
+                                        block.chunks[shard_id as usize].inner.outcome_root,
+                                        &outcome_with_proof.proof,
+                                        &outcome_with_proof.outcome.to_hashes(),
+                                    ) {
+                                        let mut chain_store_update =
+                                            self.chain.mut_store().store_update();
+                                        chain_store_update
+                                            .save_outcome_with_proof(id, outcome_with_proof);
```
Comment (SkidanovAlex) [inline chain/client/src/view_client.rs]: where do you verify that `id` corresponds to `outcome_with_proof`?

File: pytest/tests/sanity/rpc_query.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# The test launches two validating node and two observers
```
Comment (SkidanovAlex) [inline pytest/tests/sanity/rpc_query.py]: add to `nightly/nightly.txt`

File: chain/client/src/view_client.rs (hunk 1 above)
Comment (bowenwang1996) [reply chain/client/src/view_client.rs]: Why is this a problem? It is fine if we send an extra request to some other peer.

**Pull request merged**